aocdbc/day_4_passport_processing.py

```diff
@@ -0,0 +1,107 @@
+from re import fullmatch
+import sys
+from typing import Dict, List, Optional
+
+from icontract import require, ensure
+
+_REQUIRED_KEYS = set(["byr", "iyr", "eyr", "hgt", "hcl", "ecl", "pid"])
+
+
+@ensure(lambda: True)
+def blank_line_split(text: str) -> List[str]:
+    # NOTE: I tried to do a lookahead for the 2nd newline, but split doesn't take
+    # a regex.
+    return text.split("\n\n")
+
+
+@ensure(lambda: True)
+def parse(passport: str) -> Optional[Dict[str, str]]:
+    result = {}
```

Comment:
```suggestion
    result = dict()
```

Reply:
hm! Not my choice, (also slower?) but us old python 2 fogeys are used to curlys meaning dictionaries. Shrug; changed!